sdcm/nemesis.py

```diff
@@ -3085,6 +3085,12 @@ def remove_node():
             # dead_nodes_list, so the health validator terminate the job
             if exit_status != 0:
                 self.log.error(f"nodetool removenode command exited with status {exit_status}")
+                # check and clean difference between group0 and token ring,
+                garbage_host_ids = self.cluster.diff_token_ring_group0_members(verification_node)
+                self.log.debug("Difference between token ring and group0 is %s", garbage_host_ids)
+                if garbage_host_ids:
+                    self.cluster.clean_group0_garbage(verification_node)
```

Comment:
why not to check diff token ring members when exit_status == 0?
As I understand, if we remove node without issues, there should be no garbage_host_ids, and I think we should raise error event in that case, otherwise if exit_status != 0 we can skip raising error and just clean garbage.
Maybe this is the reason of https://github.com/scylladb/scylladb/issues/13379?

Reply:
I think it is not related to that issue. 
You right, if exit_code == 0, the operation finished successfull, and there is no garbage. In this we don't need to do anything, just add new new and rebuild.
But if exit_status !=0, in this case we could have some garbage and should clean, but if clean will be successfull we don't need do anything, just continue as above, for that we should n't remove node from dead node list and no need to raise error, just logged (why removenode failed). if clean garbage failed, and garbage left in group0, then we raise an issue and remove from dead_node list, so ClusterHealth validator raise critical issue

Reply:
> if exit_code == 0, the operation finished successfull, and there is no garbage

I was wondering if we can be sure of that statement - what in a case of rc==0 but something unnoticed failed in Scylla's inner workings. Is there some problem with checking also in that case?

Reply:
i think we have another mechanism to detect that. ClusterHealthValidator, DB log err events.